pytorch_pfn_extras/training/extensions/log_report.py

```diff
@@ -12,8 +12,22 @@
     _pandas_available = False
 
 
-def log_writer_save_func(target, file_o):
-    file_o.write(bytes(json.dumps(target, indent=4).encode('ascii')))
+class LogWriterSaveFunc:
+
+    def __init__(self, format):
+        self._format = format
+
+    def __call__(self, target, file_o):
+        if self._format == 'json':
+            log = json.dumps(target, indent=4)
+        elif self._format == 'json-lines':
+            log = '\n'.join([json.dumps(x) for x in target])
```

Comment:
Could you support appending the result so that only diff is written when in append=True mode?